### ISSUE TITLE
building M4 1.4.17 with GCC 4.8.2 fails on SL5

### ISSUE THREAD

**Issue by boegel** [hpcugent/easybuild-easyconfigs#529]
```
builtin.o: In function `stpncpy':
builtin.c:(.text+0x3d20): multiple definition of `stpncpy'
m4.o:m4.c:(.text+0x180): first defined here
builtin.o: In function `gets':
builtin.c:(.text+0x3d30): multiple definition of `gets'
m4.o:m4.c:(.text+0x190): first defined here
builtin.o: In function `fgets':
builtin.c:(.text+0x3d40): multiple definition of `fgets'
m4.o:m4.c:(.text+0x1a0): first defined here
...
```

side notes:
 * building `M4/1.4.17-GCC-4.8.2` works fine on SL6
 * building `M4/1.4.16-GCC-4.8.2` works fine on SL5/SL6

**Comment by boegel** [comment]
This can be fixed with http://stackoverflow.com/questions/15285776/multiple-definitions-when-linking-collect2-error-ld-returned-1-exit-status

cc @SimonPinches

**Comment by boegel** [comment]
fixed in #2774, #2779, #2816
